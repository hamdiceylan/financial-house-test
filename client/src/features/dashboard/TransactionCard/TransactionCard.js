import React from 'react'
import { Grid, Header } from 'semantic-ui-react'

const TransactionCard = ({transaction}) => {
  return (
    <Grid.Column width={8}>
        <Header icon='list' content='Transaction Details'/>
        <p>Transaction Id: <strong>{transaction.transactionId}</strong></p>
        <p>Fx TransactionId: <strong>{transaction.fxTransactionId}</strong></p>
        <p>ReferenceNo: <strong>{transaction.referenceNo}</strong></p>
        <p>Operation: <strong>{transaction.operation}</strong></p>
        <p>Status: <strong>{transaction.status}</strong></p>
        <p>Type: <strong>{transaction.type}</strong></p>
    </Grid.Column>
  )
}

export default TransactionCard
