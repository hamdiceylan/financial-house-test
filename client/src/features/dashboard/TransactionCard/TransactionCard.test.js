import React from 'react'
import { shallow } from 'enzyme'
import TransactionCard from './TransactionCard';

describe('TransactionCard', () =>{
    const props = {
        transaction : {
            transactionId: 'transactionId',
            fxTransactionId: 'fx',
            referenceNo: 'reference',
            operation: 'operation',
            status: 'status',
            type: 'type'
        }
    }
    const transactionCard = shallow(<TransactionCard {...props} />);

    it('renders correctly', () => {
        expect(transactionCard).toMatchSnapshot();
    })

})

