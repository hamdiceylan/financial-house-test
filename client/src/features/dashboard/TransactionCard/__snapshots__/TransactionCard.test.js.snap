// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionCard renders correctly 1`] = `ShallowWrapper {}`;
